packages/color-scale-block/src/ColorScaleBlock.tsx

```diff
@@ -0,0 +1,520 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import 'tailwindcss/tailwind.css';
+import { MouseEvent, useEffect, useRef, useState } from 'react';
+import { AppBridgeBlock, useBlockSettings, useColorPalettes, useEditorState } from '@frontify/app-bridge';
+import { SquareWithColor } from './components/SquareWithColor';
+import { ColorPickerFlyout } from './components/ColorPickerFlyout';
+import { ColorProps, Settings } from './types';
+import {
+    Button,
+    ButtonEmphasis,
+    ButtonGroup,
+    ButtonSize,
+    ButtonStyle,
+    ButtonType,
+    Color,
+    DropZonePosition,
+    IconArrowStretchBox12,
+    IconPlus12,
+    IconSize,
+    IconTrashBin,
+    Palette,
+    useMemoizedId,
+} from '@frontify/fondue';
+import '@frontify/fondue-tokens/styles';
+import { joinClassNames } from '@frontify/guideline-blocks-shared';
+import { DndProvider } from 'react-dnd';
+import { HTML5Backend } from 'react-dnd-html5-backend';
+import 'tailwindcss/tailwind.css';
+import { DropZone } from './dragAndDrop/DropZone';
+import { EmptyView } from './components/EmptyView';
+import {
+    DEFAULT_COLOR_SQUARE_WIDTH,
+    DROP_ZONE_WIDTH,
+    calculateDefaultColorWidth,
+    calculateWidths,
+    canExpandColorBlock,
+    getMinimumAmountOfSquares,
+    resizeEvenly,
+} from './helpers';
+
+type Props = {
+    appBridge: AppBridgeBlock;
+};
+
+export const ColorScaleBlock = ({ appBridge }: Props) => {
+    const { colorPalettes: appBridgePalettes } = useColorPalettes(appBridge);
+    const [colorPickerPalettes, setColorPickerPalettes] = useState<Palette[]>([] as Palette[]);
+    const isEditing = useEditorState(appBridge);
+    const [blockSettings, setBlockSettings] = useBlockSettings<Settings>(appBridge);
+    const [colorScaleHeight, setColorScaleHeight] = useState(
+        blockSettings.customHeight ? blockSettings.heightInput : blockSettings.heightSlider
+    );
+    const [currentlyDraggedColorId, setCurrentlyDraggedColorId] = useState<Nullable<number>>();
+    const [editedColor, setEditedColor] = useState<Nullable<ColorProps>>(null);
+    const [isColorPickerOpen, setIsColorPickerOpen] = useState<boolean>(false);
+    const colorScaleBlockRef = useRef<Nullable<HTMLDivElement>>(null);
+    const draggingId = useRef<number | null | undefined>();
+    const dragStartPos = useRef<number | null | undefined>();
+    const dragStartWidth = useRef<number | null | undefined>();
+    const lastDragPos = useRef<number | null | undefined>();
+    const timerToUpdateBlockSettings = useRef<ReturnType<typeof setTimeout>>();
+
+    const deleteColor = (id: number) => {
+        if (!id) {
+            return;
+        }
+
+        const reorderedList = displayableItems?.filter((item) => item.id !== id);
```

Comment:
Handler definitions come after hooks and state! How are you not getting warnings for using variables before they are defined? Move all state & hooks up.

Reply:
Set displaybleItems initially to [] and you don't need to check for nullish values.